client-local/src/main/java/com/cloudera/livy/client/local/LocalClient.java

```diff
@@ -339,21 +326,14 @@ public void run() {
       if (userJars != null) {
         String allJars = Joiner.on(",").join(livyJars, userJars);
         conf.set(SPARK_JARS_KEY, allJars);
-      } else {
-        argv.add("--jars");
-        argv.add(livyJars);
       }
 
-      argv.add(jar);
-      argv.add("--remote-host");
-      argv.add(serverAddress);
-      argv.add("--remote-port");
-      argv.add(serverPort);
-      argv.add("--config-file");
-      argv.add(livyConf.getAbsolutePath());
+      launcher.addAppArgs("--remote-host" + serverAddress);
+      launcher.addAppArgs("--remote-port" + serverPort);
+      launcher.addAppArgs("--config-file" + livyConf.getAbsolutePath());
+      launcher.setAppName("ClientID");
```

Comment:
The previous code did not set an app name, so the new one shouldn't either, especially not a hardcoded one that users cannot override.